server.js

```diff
@@ -1,116 +1,185 @@
+      return response.status(422).send({
+        error: `Expected format: { 'WAVE_ID': <String>, 'SOURCE_ID': <String>, 'YEAR': <String>, 'MONTH': <String>, 'LOCATION': <String>, 'MAXIMUM_HEIGHT': <String>, 'FATALITIES': <string>, 'FATALITY_ESTIMATE': <String>, 'ALL_DAMAGE_MILLIONS': <String>, 'DAMAGE_ESTIMATE': <String> }. You're missing a ${requiredParameter} property.`,
+      });
+    }
+  }
+
+  database('waves')
+    .insert(wave, '*')
     .then(wave => response.status(201).json(wave))
-    .catch(error => response.status(500).json({error}));
+    .catch(error => response.status(500).json({ error }));
 });
 
 app.delete('/api/v1/sources/:id', (request, response) => {
   const { id } = request.params;
 
-  database('waves').where({SOURCE_ID: id}).del().then(() => {
-    database('sources').where({SOURCE_ID: id}).del()
-    .then(deleted => !deleted ? response.status(404).json({error: "Could not be found."}) : response.sendStatus(204))
-    .catch(error => response.status(500).json({error}));
-  })
+  database('waves')
+    .where({ SOURCE_ID: id })
+    .del()
+    .then(() => {
+      database('sources')
+        .where({ SOURCE_ID: id })
+        .del()
+        .then(deleted =>
+          (!deleted
+            ? response.status(404).json({ error: 'Could not be found.' })
+            : response.sendStatus(204)))
+        .catch(error => response.status(500).json({ error }));
+    });
 });
 
 app.delete('/api/v1/waves/:id', (request, response) => {
   const { id } = request.params;
 
-  database('waves').where({WAVE_ID: id}).del()
-    .then(deleted => !deleted ? response.status(404).json({error: "Could not be found."}) : response.sendStatus(204))
-    .catch(error => response.status(500).json({error}));
+  database('waves')
+    .where({ WAVE_ID: id })
+    .del()
+    .then(deleted =>
+      (!deleted
+        ? response.status(404).json({ error: 'Could not be found.' })
+        : response.sendStatus(204)))
+    .catch(error => response.status(500).json({ error }));
 });
 
 app.patch('/api/v1/sources/:id', (request, response) => {
+  const sourcePatch = request.body;
   const { id } = request.params;
-  const propUpdate = request.body;
-  const propKeys = Object.keys(propUpdate);
-  const prop = propKeys[1]
-  const updateValue = propUpdate[prop]
 
-  database('sources').where({SOURCE_ID: id}).update({prop: updateValue}).then(updated => !updated ? response.status(404).json({error: 'Must receive SOURCE_ID and the prop to update'}) : response.sendStatus(200))
-  .catch(error => response.status(500).json({error}))
-})
+  database('sources')
+    .where('SOURCE_ID', id)
+    .update(sourcePatch, '*')
+    .then((update) => {
+      if (!update.length) {
+        response.status(404).json({
+          error: `Cannot find Source with ID of ${id}`,
+        });
+      }
+      response.status(200).json({ patchedSource: update[0] });
+    })
+    .catch(error => response.status(500).json({ error }));
+});
```

Comment:
Here is the updated PATCH method. I use the .where to select the source that has the same SOURCE_ID as the id passed in to the URL. Then I updated selected object with the property that is being passed in. Then I did the data handling and returned the whole patched object if successful